### PR TITLE
Enable Let's Encrypt automation for k8s.io redirector

### LINKED ISSUE #215: move k8s.io redirector behind Ingress and automate TLS using cert-manager
The TLS cert for the k8s.io redirector expires in about a week (April 11). Let's get it automated!

I've already got cert-manager set up in the cluster, so now it's just a matter of getting the redirector running behind an `Ingress` without breaking anything.

Steps to get there:
1. Teach nginx how to handle forwarded HTTPS requests from the GCP Ingress (since it won't terminate HTTPS directly in this case).
2. Create an `Ingress` to the existing `LoadBalancer` service, using a new static IP as the existing static IP is still bound to the `LoadBalancer`. Use the existing soon-to-expire TLS cert to terminate HTTPS on the `Ingress`.
3. Update DNS records to point to the new static IP and wait for propagation.
4. Switch the service to `NodePort`, releasing the old static IP.
5. Create cert-manager `Certificate` resources - use the Let's Encrypt staging endpoint for canary, and prod endpoint for prod. (Note: because of the large number of subdomains, this creates a lot of backend services at once (~50). We'll need to increase our backend service quota first.)
6. After the TLS cert has been created (may take some time), update the `Ingress` to use the new secret. (I suggest using a new name here, since we'll be using different Let's Encrypt endpoints for canary and prod.)
7. Switch off HTTPS support in nginx and delete the old certificate.
8. Hopefully never think about this again. :)

/assign
cc @thockin @munnerz @nikhita

### PR DESCRIPTION
**Pull request by ixdy**
OK, the last sequence of changes to fix #215.

Commits, in order they should be applied:
1. Add a self-signing dummy `ClusterIssuer`. We'll use this for the canary instance of the redirector, since we can't use Let's Encrypt HTTP challenges since the domains don't point at the canary Ingress.
2. Remove and fix old sites on the redirector. I took this from #218. These are either subdomains that no longer point at the redirector, or are redirections we were doing wrong.
3. Create `Certificate` resources which will generate TLS certificates (saving them into a secret named `k8s-io-tls`). I manually created these lists by looking at the DNS configuration for subdomains that point at the redirector. TODO: better management of these lists.
4. After the new TLS secrets have been created (will take some time), point the `Ingress` at them.
  - Note that I already bumped up our backend services quota on the GCP project to 100.
5. Switch service to `NodePort`, since we can use the `Ingress` exclusively now.
6. Stop serving HTTPS from nginx. (I'll also delete the old secret at this point.)

### CODE REVIEW

**Comment by thockin** [comment]
There's a lot in here, but it looks OK.  What is "dev" ?  I have no memory of that being added.  What is it?

**Comment by ixdy** [comment]
it appears to be what gets deployed if you run `make deploy` - I guess if you're testing changes on your own kubernetes cluster? It was added in https://github.com/kubernetes/k8s.io/pull/7. 

Admittedly I've just been using the canary configs to test on my own cluster. I can delete `dev` if you'd prefer.

**Comment by ixdy** [comment]
I'm going to go ahead and deploy these changes to canary.

**Comment by ixdy** [comment]
Canary deployment is working and passing all tests.

File: k8s.io/Makefile [hunk 1]
```diff
@@ -14,6 +14,7 @@ deploy:
 	kubectl apply -f configmap-www-get.yaml
 	kubectl apply -f service-dev.yaml
 	kubectl apply -f deployment.yaml
+	kubectl apply -f certificate-dev.yaml
```
Comment (thockin) [inline k8s.io/Makefile]: I don't think anyone uses the -dev variants.  Can we just nuke them?  In fact we might want to nuke this rule - I forgot it existed

Comment (ixdy) [reply k8s.io/Makefile]: deleted!

**Comment by ixdy** [comment]
@thockin any other comments?

**Comment by cblecker** [comment]
/lgtm
/approve
/hold

This looks sane to me.

**Comment by ixdy** [comment]
Updated ConfigMap, Deployment, and Service have been applied; tests still pass.

Applying Certificate resource now.

**Comment by ixdy** [comment]
Oops, bumped into a quota limit:
```
loadbalancer-controller  Error during sync: googleapi: Error 403: Quota 'HEALTH_CHECKS' exceeded. Limit: 50.0 globally., quotaExceeded
```

Quota bump has been requested.

**Comment by ixdy** [comment]
I deleted the certificate resource in the prod namespace. Will reattempt once quota bump has been applied.

**Comment by ixdy** [comment]
quota bump is in; trying to create a certificate again.

**Comment by ixdy** [comment]
This has now been fully deployed. Last remaining issue I found will be addressed by https://github.com/kubernetes/k8s.io/pull/224.

**Pull request merged**